### PR TITLE
Implement lint/noExplicitAny

### PR DESCRIPTION
**Pull request by KevinKelbie**
Addresses #153

I don’t think this PR is ready to merge yet but I wanted to get feedback on my approach. I’m happy with the implementation of `noExplicitAny` but I had to modify the `testLint` function to accept flow and ts syntaxes but the snapshot still reads as a javascript snapshot rather than ts and flow. Another thing was that `preferFunctionDeclarations.test.ts` has a [test](https://github.com/facebookexperimental/rome/blob/fee494f1e6bcda26707157f0b9dd6279b4e8af1b/packages/%40romejs/js-compiler/transforms/lint/preferFunctionDeclarations.test.ts#L25) which uses ts/flow syntax without using the appropriate linting so the snapshots generated are not that useful so I’ve updated them. In addition to that, I had to disable the suppressions logic for `rome test` & `rome ci` because of an issue I raised to you on discord and #154

Let me know if I'm on the right track and how I should procceed, thanks.

### CODE REVIEW

File: packages/@romejs/cli-reporter/Reporter.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ type ListOptions = {
   ordered?: boolean;
 };
 
+// rome-suppress lint/noExplicitAny lint/noExplicitAny
```
Comment (sebmck) [inline packages/@romejs/cli-reporter/Reporter.ts]:
```suggestion
// rome-suppress lint/noExplicitAny
```

File: packages/@romejs/core/master/Master.ts [hunk 2]
```diff
@@ -230,7 +230,9 @@ export default class Master {
     process.exit();
   }
 
+  // rome-suppress lint/noExplicitAny lint/noExplicitAny
```
Comment (sebmck) [inline packages/@romejs/core/master/Master.ts]:
```suggestion
  // rome-suppress lint/noExplicitAny
```

File: packages/@romejs/core/master/Master.ts [hunk 3]
```diff
@@ -230,7 +230,9 @@ export default class Master {
...
   wrapFatal<T extends (...args: Array<any>) => any>(callback: T): T {
+    // rome-suppress lint/noExplicitAny lint/noExplicitAny
```
Comment (sebmck) [inline packages/@romejs/core/master/Master.ts]:
```suggestion
    // rome-suppress lint/noExplicitAny
```

**Comment by sebmck** [comment]
Looks great! Only complaint is the duplicate category names. We should add something to prevent that. If you can fix those it looks good to merge!

**Comment by KevinKelbie** [comment]
Thanks for the review!:) The duplicate category names are intentional. It turns out if you have multiple uses of `any` on a single line you need to suppress each one individually. I wasn't sure if this was expected behaviour or not. 

If you only suppress it once it ignores the first occurrence rather than every occurrence:
Screenshot 2020-03-13 at 19 50 16

**Comment by sebmck** [comment]
Ahhhh. Yeah we remove the suppressions as we find one that matches so we can produce "unused suppression" rules. Let's merge this as-is and I'll fix that and put up a new PR.

**Comment by sebmck** [comment]
Thank you!

**Comment by sebmck** [comment]
Gah, accidentally committed to `master` as d9a1bca882e750c7f9619c5e21f5c88e37560253 and ce2178c8b99d80dff3cabd7a4ad2df002941d290. We should setup branch protection.

**Pull request merged**